ui/apps/dashboard/src/queries/restAPIHooks.ts

```diff
@@ -0,0 +1,40 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { useAuth } from '@clerk/nextjs';
+
+export function useRestAPIRequest<T>({
```

Comment:
Created this as we had not simple way to make REST API requests from the client with Clerk auth.